AWSIoTTests/AWSIoTDataManagerTests.swift

```diff
@@ -741,21 +740,18 @@ class AWSIoTDataManagerTests: XCTestCase {
 
         let testMessage = "Test Message"
         let testTopic = "TestTopic"
-
+        gotMessage.expectedFulfillmentCount = 5
         //Subscribe to TestTopic
         iotDataManager.subscribe(toTopic: testTopic, qoS: .messageDeliveryAttemptedAtLeastOnce, messageCallback: {
             (payload) ->Void in
             let stringValue:String = NSString(data: payload, encoding: String.Encoding.utf8.rawValue)! as String
             print("received: \(stringValue)")
             XCTAssertEqual(testMessage, stringValue)
-            messageCount = messageCount+1
```

Comment:
looks like the problem with the old code is that there isn't a semphore on updating messageCount so it may never reach 5 even if 5 callbacks are triggered